share/wake/lib/system/remote_cache_api.wake

```diff
@@ -533,78 +533,99 @@ export target rscApiCheckAuthorization (api: RemoteCacheApi): Result Unit Error
         Some x -> failWithError "Invalid auth key. Status code: {str x}"
         None -> failWithError "Invalid auth key. Unable to determine status code"
 
+# Converts db only blob uris into actual data
+# maps path = "%46%6F%6F" to content = "foo"
+def rscDbBlobToContent path =
+    # If path is empty we have the empty string
+    require True = path.strlen > 0
+    else Pass ""
+
+    # If not empty then it must as least be %00
+    require True = path.strlen >= 3
+    else failWithError "rsc: Invalid db path: '{path}'"
+
+    require "", byteStrs = tokenize `%` path
+    else failWithError "rsc: Failed to extract bytes from db path: '{path}'"
+
+    require Some bytes =
+        byteStrs
+        | map (intbase 16)
+        | findNone
+    else failWithError "rsc: Failed to parse bytes strings into bytes from db path: '{path}'"
+
+    bytes
+    | map integerToByte
+    | cat
+    | Pass
+
+# Wrap scheme dispatch behind uri interface
+def uriSchemeFn dbBlobFn dbFileFn: String => Result a Error =
```

Comment:
I'm not quite happy with the structure of this function. I think the ordering is a bit confusing.

Is there a reason we can't reorder it? something like

```ruby
def unsupportedScheme = ...

require scheme, path, Nil = uri | extract `...`
else failWithError "..."

match scheme
  "db" -> Fn
  ...
  _ -> unsupprtedScheme
```


The other question I have, both places calling this function want a return type of `Result String Error` is there any reason we can't just call `dbBlobfn`/`dbFileFn` on their behalf?

Reply:
I got caught up thinking we shouldn't call it because one wanted a "Path" as a string and the other wanted content as the string. But then I did all the work to make it polymorphic so there wasn't a reason not to just return the `Result a Error`. Thanks for the suggestion and fix.